src/classes/Entity.ts

```diff
@@ -1,11 +1,18 @@
+import sha256 from "crypto-js/sha256";
+
 export default class Entity {
-    public id: string;
-    public type: string;
-    public data: any;
+  public id: string;
+  public params: any;
+
+  constructor(id: string, params?: any) {
+    this.id = id;
+    this.params = params || {};
+  }
 
-    constructor(id: string, type: string, data: string) {
-        this.id = id;
-        this.type = type;
-        this.data = data;
-    }
+  public equals = (e: Entity) => {
+    return (
+      sha256(JSON.stringify(e)).toString() ===
```

Comment:
are we certain about JSON.stringify() doing deterministic ordered serialization?

Reply:
will have to read/test more on it

Reply:
I think we should go with https://www.npmjs.com/package/json-stringify-deterministic